nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/PutGCSObject.java

```diff
@@ -542,9 +544,15 @@ public void process(InputStream rawIn) throws IOException {
             }
             session.transfer(flowFile, REL_SUCCESS);
             final long millis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
-            final String url = "https://" + bucket + ".storage.googleapis.com/" + key;
 
-            session.getProvenanceReporter().send(flowFile, url, millis);
+            String transitUri;
+            try {
+                final URL url = new URL(storage.getOptions().getHost());
+                transitUri = String.format("%s://%s.%s/%s", url.getProtocol(), bucket, url.getHost(), key);
+            } catch (MalformedURLException e) {
+                transitUri = e.getClass().getSimpleName();
+            }
```

Comment:
It seems like the implementation could be moved to a protected `getTransitUri()` method in `AbstractGCSProcessor`, or the changed could be implemented in both classes.
```suggestion
        final URI storageApiUri = URI.create(storage.getOptions().getHost());
        final String transitUri = String.format("%s://%s.%s/%s", uri,getScheme(), bucketName, uri.getHost(), key);
```